subsys/bluetooth/controller/ll_sw/nordic/lll/lll_sync.c

```diff
@@ -0,0 +1,301 @@
+#include "hal/debug.h"
+
+static int init_reset(void);
+static int prepare_cb(struct lll_prepare_param *p);
+static void isr_rx(void *param);
+
+int lll_sync_init(void)
+{
+	int err;
+
+	err = init_reset();
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+int lll_sync_reset(void)
+{
+	int err;
+
+	err = init_reset();
+	if (err) {
+		return err;
+	}
+
+	return 0;
+}
+
+void lll_sync_prepare(void *param)
+{
+	struct lll_prepare_param *p;
+	struct lll_sync *lll;
+	uint16_t elapsed;
+	int err;
+
+	p = param;
+	lll = p->param;
+
+	/* Request to start HF Clock */
+	err = lll_hfclock_on();
+	LL_ASSERT(err >= 0);
+
+	/* Instants elapsed */
+	elapsed = p->lazy + 1;
+
+	/* Save the (skip + 1) for use in event */
+	lll->skip_prepare += elapsed;
+
+	/* Accumulate window widening */
+	lll->window_widening_prepare_us += lll->window_widening_periodic_us *
+					   elapsed;
+	if (lll->window_widening_prepare_us > lll->window_widening_max_us) {
+		lll->window_widening_prepare_us = lll->window_widening_max_us;
+	}
+
+	/* Invoke common pipeline handling of prepare */
+	err = lll_prepare(lll_is_abort_cb, lll_abort_cb, prepare_cb, 0, p);
+	LL_ASSERT(!err || err == -EINPROGRESS);
+}
+
+static int init_reset(void)
+{
+	return 0;
+}
+
+static int prepare_cb(struct lll_prepare_param *p)
+{
+	struct node_rx_pdu *node_rx;
+	uint32_t ticks_at_event;
+	uint32_t ticks_at_start;
+	uint16_t event_counter;
+	uint32_t remainder_us;
+	uint8_t data_chan_use;
+	struct lll_sync *lll;
+	struct evt_hdr *evt;
+	uint32_t remainder;
+	uint32_t hcto;
+
+	DEBUG_RADIO_START_O(1);
+
+	lll = p->param;
+
+	/* Deduce the skip */
+	lll->skip_event = lll->skip_prepare - 1;
+
+	/* Calculate the current event counter value */
+	event_counter = lll->event_counter + lll->skip_event;
+
+	/* Update event counter to next value */
+	lll->event_counter = lll->event_counter + lll->skip_prepare;
+
+	/* Reset accumulated latencies */
+	lll->skip_prepare = 0;
+
+	/* Current window widening */
+	lll->window_widening_event_us += lll->window_widening_prepare_us;
+	lll->window_widening_prepare_us = 0;
+	if (lll->window_widening_event_us > lll->window_widening_max_us) {
+		lll->window_widening_event_us =	lll->window_widening_max_us;
+	}
+
+	/* Calculate the radio channel to use */
+	data_chan_use = lll_chan_sel_2(event_counter, lll->data_chan_id,
+				       &lll->data_chan_map[0],
+				       lll->data_chan_count);
+
+	/* Start setting up Radio h/w */
+	radio_reset();
+
+#if defined(CONFIG_BT_CTLR_TX_PWR_DYNAMIC_CONTROL)
+	radio_tx_power_set(lll->tx_pwr_lvl);
+#else
+	radio_tx_power_set(RADIO_TXP_DEFAULT);
+#endif /* CONFIG_BT_CTLR_TX_PWR_DYNAMIC_CONTROL */
+
+	radio_phy_set(lll->phy, 1);
+	radio_pkt_configure(8, PDU_AC_PAYLOAD_SIZE_MAX, (lll->phy << 1));
+
+	node_rx = ull_pdu_rx_alloc_peek(1);
+	LL_ASSERT(node_rx);
+
+	radio_pkt_rx_set(node_rx->pdu);
+
+	radio_aa_set(lll->access_addr);
+	radio_crc_configure(((0x5bUL) | ((0x06UL) << 8) | ((0x00UL) << 16)),
+			    (((uint32_t)lll->crc_init[2] << 16) |
```

Comment:
I get use sys_get_le24. But will do in a separate PR, so that I can change other occurrences. No action in this PR.

Issue created: https://github.com/zephyrproject-rtos/zephyr/issues/27971

Reply:
Yes, a separate PR and fixing other occurrences is a good idea